### ISSUE TITLE
Allow manual adjustment of FOV & view direction

### ISSUE THREAD

**Issue by nikcain** [ShowMeHills/ShowMeHills#22]
Apparently some phones don't have compasses (!) so as a workaround the FOV and view could be handled manually. This could also be really useful for anyone struggling to get the compass driven FOV & view accurate.

Need to add pinch to adjust FOV, and then swipe to change view direction.
Probably need to have a setting to lock actions down so that it can't happen accidentally.

**Comment by tjsandberg** [comment]
Would love to see this implemented because my cheap Galaxy Core Prime from T-Mobile doesn't have a compass. I have not coded for Android before and I'm pretty new to Github too, but I'd be willing to help if someone could point me in the right direction.

Tom

**Comment by nikcain** [comment]
I've just started adding in some new features (adding sliders on the main view to filter distance and mountain height), so will definitely consider getting this in as well. 

If you want to try and do some coding you can just download it as a zip (https://github.com/ShowMeHills/ShowMeHills/archive/master.zip), then just install Android Studio and see if you can get it to build (hopefully I've not missed any files out) - from Android Studio you'd open the project in ShowMeHills-master\ShowMe\ShowMeHills

There's a little bit of code already in there to handle multitouch, although it's commented out at the moment - I managed to pick up the events, but not tie them into the FOV value

**Comment by tjsandberg** [comment]
Thanks for the response. I'm installing Android Studio and its dependencies. I did a fork then a clone of the ShowMeHills repository. Is that the same as downloading the master.zip file?

**Comment by nikcain** [comment]
Yes, that'll be fine. You only needed to clone it really - clone and fork are pretty much the same thing, except fork means you have your own github repository. Any changes or additions you make can be pushed up to your repository without affecting this one. You can keep your repo up to date with this one as well, although it's not automatic.

Let me know if you have any problems - I'm happy to help (we could switch to email rather than use the comments here).

**Comment by tjsandberg** [comment]
Sorry, I got sidetracked from this for a while. Not 100% sure when I will get back to it, or how much time it will take me to understand the code well enough to be useful.

Meanwhile I bought a Nexus 5X which does have the compass. Unfortunately it also has a hardware "feature" that causes the camera image to be upside down in ShowMeHills and many other apps. Much discussion and info here:
[https://www.reddit.com/r/Android/comments/3rjbo8/nexus5x_marshmallow_camera_problem/](url)

So I have even more motivation to get moving on this.

Wasn't quite sure how to take this discussion to email. Several addresses show up when I do reply all.

Thanks,
Tom

**Comment by nikcain** [comment]
Ah, someone else reported that - I suggested there was an autorotate
setting in the main phone settings (under Settings - Accessibility), and it
needed to be on. They didn't get back to me so I thought it was ok, but
that thread does make it look like it's something else.

One of the replies has a link to the android documentation which has a
snippet of code that might help (
https://developer.android.com/reference/android/hardware/Camera.html#setDisplayOrientation(int)
). Let me know if the settings thing helps, or doesn't make difference, and
if it's no good I'll do an update on ShowMeHills.

Don't worry about the email addresses - it looks like this is all coming
through github. If you want to write to me without it showing up publicly
on github just use nik@showmehills.com

Nik

**Comment by tjsandberg** [comment]
Changing the autorotate setting didn't help.

I see the same upside down issue in my banking app when I try to take a picture of a check. Looks like it is the LG/Android screw-up as described in the thread.

**Comment by nikcain** [comment]
Ok, I'll have a good read of the thread. Maybe at the very least I can find
a manual workaround if there isnt a general fix.

Nik

**Comment by tjsandberg** [comment]
Hi Nik,
Hope all is well with you. My Nexus 5 just upgraded to Android 7, and the upside down camera problem is fixed, both in ShowMeHills and my banking app. :-)
Tom

**Comment by nikcain** [comment]
Hi Tom,

That's great!

By coincidence I've a version in final testing that I want to release any day now which updates the camera library it uses ~ sounds like it's not needed now :-) However I've also added some extra controls to filter out hills so should still be worth it!

Nik
